### ISSUE TITLE
"pageContext.urlParsed.hashOriginal isn't available on the server-side" warning

### ISSUE THREAD

**Issue by chrisvander** [brillout/vite-plugin-ssr#1092]
### Description

I've recently upgraded to the latest version and am being inundated with messages like "pageContext.urlParsed.hashOriginal isn't available on the server-side" on almost every render. I get it, hashOriginal isn't server side. The problem is I never actually reference `hashOriginal`, not client nor serverside. I am, however, using the search params in a query hook (persist some state through refresh in the query), but I'm not sure why it's being triggered:
```ts
import { useState } from "react"
import { usePageContext } from "../renderer/usePageContext"

export function useQueryParams<T extends number | string>(key: string, defaultItem: T) {
  const { urlParsed } = usePageContext()
  const initialFromUrl = urlParsed?.search[key]
  const initialValue = (
    typeof defaultItem === "number" && initialFromUrl ? parseInt(initialFromUrl) : initialFromUrl
  ) as T | undefined

  const [value, setInnerValue] = useState<T>(initialValue || defaultItem)
  const setValue = (newValue: T) => {
    const newUrl = new URL(window.location.href)
    newUrl.searchParams.set(key, typeof newValue === "number" ? newValue.toString() : newValue)
    setInnerValue(newValue)
    window.history.pushState({}, "", newUrl.toString())
  }

  return [value, setValue] as const
}
```

I'm guessing this error is due to the fact that, for the hook, I'm passing `pageContext.urlParsed` to the client and serialization is triggering this. Can we use some smarter types in lieu of this message?

**Comment by brillout** [comment]
Are you using Server Routing or Client Routing?

**Comment by chrisvander** [comment]
Server routing

**Comment by brillout** [comment]
> smarter types 

Good idea. Done.

> in lieu of this message

I kept the warning but it shouldn't false-postively trigger anymore.

Released in `0.4.138`. (FYI `PageContextBuiltIn` is now named `PageContextBuiltInServer`.)

Btw. in case your company is up for it: https://github.com/sponsors/brillout.
